lib/ansible/modules/network/nxos/nxos_bfd_global.py

```diff
@@ -0,0 +1,308 @@
+    '''Ensure BFD feature is enabled. Create cmd_ref dict.
+    '''
+    output = execute_show_command(module, 'show run bfd', 'text')
+    if output and 'CLI command error' in output:
+        if module.check_mode:
+            msg = "** 'feature bfd' is disabled. BFD will auto-enable when not in check_mode **"
+            module.warn(msg)
+        else:
+            load_config(module, 'feature bfd')
+
+    cmd_ref = yaml.load(BFD_CMD_REF)
+    cmd_ref = get_platform_defaults(module, cmd_ref)
+    return cmd_ref
+
+def get_platform_defaults(module, cmd_ref):
+    '''Get platform type and update cmd_ref with platform specific defaults
+    '''
+    cap = get_capabilities(module)
+    if cap:
+        device_info = cap['device_info']['network_os_platform']
+        plat = None
+        if 'N3K' in device_info:
+            plat = 'N3K'
+        if plat:
+            for k in cmd_ref:
+                if plat in cmd_ref[k]:
+                    cmd_ref[k]['default'] = cmd_ref[k][plat]
+
+    return cmd_ref
+
+
+def execute_show_command(module, command, format):
+    cmds = [{
+        'command': command,
+        'output': format,
+    }]
+    output = None
+    try:
+        output = run_commands(module, cmds)[0]
+    except ConnectionError as exc:
+        if 'CLI command error' in repr(exc):
+            output = repr(exc)
+        else:
+            raise
+    return output
+
+
+def get_existing(module, cmd_ref, show_cmd):
+    '''
+    Get a list of existing command states from device; then update cmd_ref
+    with any 'existing' values that differ from default states
+    '''
+    output = execute_show_command(module, show_cmd, 'text') or []
+    if not output:
+        return {}
+
+    # Walk each cmd in cmd_ref, use cmd pattern to discover existing cmds
+    output = output.split('\n')
+    for k in cmd_ref:
+        pattern = cmd_ref[k]['getval']
+        match = [m.groups() for m in (re.search(pattern, line) for line in output) if m]
+        if not match:
+            continue
+        match = list(match[0]) # tuple to list
+        # handle 'no' keyword
+        if None is match[0]:
+            match.pop(0)
+        elif 'no' in match[0]:
+            cmd_ref[k]['no_cmd'] = True
+            match.pop(0)
+            if not match:
+                continue
+        type = cmd_ref[k]['type']
+        if 'int' == type:
+            cmd_ref[k]['existing'] = int(match[0])
+        if 'int_list' == type:
+            cmd_ref[k]['existing'] = [int(i) for i in match]
+        if 'str' == type:
+            cmd_ref[k]['existing'] = match[0]
+
+    logit(cmd_ref)
+    return cmd_ref
+
+
+def get_playvals(module, cmd_ref):
+    '''Update cmd_ref with playbook values
+    '''
+    for k in cmd_ref.keys():
+        if k in module.params and module.params[k] is not None:
+            playval = module.params[k]
+            if 'int' is cmd_ref[k]['type']:
+                playval = int(playval)
+            elif 'int_list' is cmd_ref[k]['type']:
```

Comment:
I think these checks should be using `==` rather than `is`.

Reply:
Agreed.